From 17b2a2efa243311b5cee9d4a51029d292d7e2483 Mon Sep 17 00:00:00 2001
From: Toni Heidenreich <tonihei@google.com>
Date: Tue, 8 Jul 2025 03:46:04 -0700
Subject: [PATCH] Do not propagate WIU/BFSL for MediaController.sendCommand

This call is just for exchanging technical information and
shouldn't be used to grant another app permission to start a
foreground service or gain while-in-use permissions.

Manual cherry-pick of ag/33211646

Bug: 339637822
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:37de99f4813e3dd027412720a99721bb43828154)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:e4c59cb5110d47e3f93488c1a684f5ca22a40c8b)
Merged-In: I7cedaa5a17501b8e5d4517cf2080d3cb3f4ee8ea
Change-Id: I7cedaa5a17501b8e5d4517cf2080d3cb3f4ee8ea
---
 .../core/java/com/android/server/media/MediaSessionRecord.java | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/services/core/java/com/android/server/media/MediaSessionRecord.java b/services/core/java/com/android/server/media/MediaSessionRecord.java
index 1b2f2beb2104..c36a059e4da6 100644
--- a/services/core/java/com/android/server/media/MediaSessionRecord.java
+++ b/services/core/java/com/android/server/media/MediaSessionRecord.java
@@ -1246,9 +1246,6 @@ public class MediaSessionRecord implements IBinder.DeathRecipient, MediaSessionR
         public void sendCommand(String packageName, int pid, int uid, String command, Bundle args,
                 ResultReceiver cb) {
             try {
-                final String reason = TAG + ":" + command;
-                mService.tempAllowlistTargetPkgIfPossible(getUid(), getPackageName(),
-                        pid, uid, packageName, reason);
                 mCb.onCommand(packageName, pid, uid, command, args, cb);
             } catch (RemoteException e) {
                 Log.e(TAG, "Remote failure in sendCommand.", e);
-- 
2.52.0.rc1.455.g30608eb744-goog

